### PR TITLE
fix(hooks): don't block a merge on a review finding that lands on a doc file

### PR DESCRIPTION
**Pull request by WingedGuardian**
### The problem
`git_push_guard._check_inline_review_findings` (the merge gate that scans Codex
INLINE review comments) blocked a merge on ANY unresolved `[P1]`. A `[P1]`
anchored to a documentation file — a CHANGELOG typo, a README wording nit — is
not a code defect, yet it blocked the merge.

### The change
Add a **fail-closed** documentation-path allowlist. A finding whose file `.path`
is provably prose is surfaced to stderr but does not block:
- a `CHANGELOG`/`README`/`LICENSE`/`NOTICE`-named file (doc or no extension),
- any doc-extension file (`.md`/`.rst`/`.txt`/…) under a top-level `docs/`,
- any `*.rst` anywhere.

Everything else still blocks — a random top-level `*.md` (`NOTES.md`), ANY
source/config/executable file even under `docs/` (`docs/conf.py`, `docs/build.rs`,
`docs/config.yaml`), a missing path, or a path bearing a control character.

Only the inline endpoint (`pulls/N/comments`) carries a per-file path; the
PR-level review-body gate (`issues/N/comments`) has no file attribution and is
unchanged. The fail-closed unreadable/incomplete-read logic, the maintainer-reply
engagement check, and the P2 path are untouched.

### Safety
An allowlist, not a denylist: a code-extension denylist can't enumerate every
source/config type, so the unknown fails closed (blocks). Reviewed by
genesis-security-reviewer — an initial denylist draft was rejected for that
enumeration gap; the allowlist + control-char guard close both findings
(HIGH: non-prose under `docs/`; LOW: trailing-newline `.match` leak).

### Tests
`tests/test_hooks/test_merge_review_gate.py` — verify-RED confirmed: doc-path
P1s no longer block; non-prose-under-`docs/` and control-char paths still block;
a mixed doc+code PR blocks and names only the code finding; the jq projection is
locked.

Ledger: 54eb3752074d4afe9ccca244353ede8a

### CODE REVIEW

**Comment by WingedGuardian** [comment]
@codex review

**Comment by WingedGuardian** [comment]
**Automated first-pass review** (structural + PII/secrets/wording)

## Structural Review

No ERROR or WARNING findings.

- No pattern violations (no `nohup`, no raw SQL, no bare `except:`, no mutable default args, no `git add -A`/`--force`, no `/tmp/` usage, no direct `task_states` writes).
- Import hygiene clean — no new imports added.
- Test coverage is strong: `_is_doc_path` and the doc-path branch of `_check_inline_review_findings` get 15 new test cases in `tests/test_hooks/test_merge_review_gate.py`, including fail-closed edge cases (non-prose under `docs/`, control-char path, missing path, mixed doc+code PR).
- Commit/PR title carries a conventional prefix (`fix(hooks): ...`).

**NOTE** (documentation) — `scripts/hooks/git_push_guard.py:1104` — `_check_inline_review_findings`'s docstring ("P1 findings block unless their thread has a MAINTAINER reply ... or the merge carries `# review-override`") wasn't updated to mention the new doc-path exemption added by this PR. The behavior is well-documented in the surrounding code comments (`_is_doc_path` docstring, the `_DOC_EXTS`/`_DOC_STEMS` section comment), just not in the function's own docstring — worth a one-line addition for anyone reading the function signature in isolation. Non-blocking.

Logic spot-check on `_is_doc_path` (scripts/hooks/git_push_guard.py:~903): allowlist correctly fails closed on `CHANGELOG.py`-style disguised code, nested non-top-level `docs/` dirs, uppercase `DOCS/`, and control-char/newline-bearing paths — matches the stated fail-closed design and the security-review findings referenced in the PR description.

## PII/Secrets/Wording Scan

CLEAN — no API keys, tokens, private IPs, credentials, personal emails, hardcoded usernamed paths, or other PII/secrets in the added lines. PR title, body, and the CHANGELOG.md addition describe the fix in generic terms (merge-gate behavior, doc-path allowlist) with no security-incident language or exploitable specifics.




---
_Generated by [Claude Code](https://claude.ai/code)_

File: scripts/hooks/git_push_guard.py [hunk 1]
```diff
@@ -865,6 +865,54 @@ def _pr_ci_status(pr_num: str, repo: str | None = None) -> tuple[str, list[str]]
...
+    ext = ext.lower() if dot else ""
+    stem_l = (stem if dot else base).lower()
+    # (1) A known doc-named file with a doc/empty extension, at any depth.
+    if stem_l in _DOC_STEMS and ext in _DOC_EXTS | {""}:
+        return True
+    # (2) A pure documentation format (reStructuredText) anywhere.
+    if ext == "rst":
+        return True
+    # (3) A documentation-extension file under a top-level docs/ directory.
+    return ext in _DOC_EXTS and path.startswith("docs/")
```
[the remark this answers is not in the document]
Comment (WingedGuardian) [reply scripts/hooks/git_push_guard.py:922]: Fixed in 4c8236be. Dropped `.txt` from the blanket doc-extension set; it is now treated as prose only on a known doc-named stem (LICENSE.txt / README.txt), so docs/requirements.txt, docs/CMakeLists.txt, docs/constraints.txt, docs/meson_options.txt and a plain docs/*.txt now block. Regression tests added.

File: scripts/hooks/git_push_guard.py [hunk 2]
```diff
@@ -1101,10 +1150,24 @@ def _check_inline_review_findings(
         if _INLINE_P1_RE.search(body):
             if c.get("id") in replied_to:
                 continue  # thread engaged — treated as acknowledged
+            if _is_doc_path(c.get("path") or ""):
+                # A P1 on a documentation file (ledger 54eb3752) is surfaced but
+                # does NOT block; a code file (incl. code under docs/) still does.
+                doc_skipped.append(_inline_title(body))
+                continue
```
[the remark this answers is not in the document]
Comment (WingedGuardian) [reply scripts/hooks/git_push_guard.py:1166]: Acknowledged and consciously accepted as a design decision. This gate is a mechanical path classifier; reliably separating a wording nit from a substantive runbook error needs semantic judgment it cannot do. Crucially the finding is not suppressed — every doc-path P1 is still surfaced to stderr for the author to see; it simply does not hard-block the merge, matching this feature's path-based design under a single-author threat model. Keeping the broad exemption.

**Comment by WingedGuardian** [comment]
@codex review

File: scripts/hooks/git_push_guard.py [hunk 3]
```diff
@@ -865,6 +865,54 @@ def _pr_ci_status(pr_num: str, repo: str | None = None) -> tuple[str, list[str]]
...
+    "contributing",
+}
+
+
+def _is_doc_path(path: str) -> bool:
+    """Whether a review finding's file *path* is documentation whose findings do
+    NOT block a merge. FAIL-CLOSED allowlist (see the section comment): anything
+    not provably prose — a source/config file under ``docs/``, a random top-level
+    ``*.md``, a missing path, or a control-char-bearing path — blocks."""
+    if not path or any(ord(c) < 0x20 for c in path):
```
[the remark this answers is not in the document]
Comment (WingedGuardian) [reply scripts/hooks/git_push_guard.py]: Fixed in 4c8236be — the control-char guard now rejects the complete Unicode Cc range (C0 <0x20, DEL 0x7F, and C1 0x80-0x9F incl. NEL), not just C0. Regression test added.
